packages/node_modules/node-red/red.js

```diff
@@ -130,6 +155,33 @@ try {
     process.exit();
 }
 
+if (parsedArgs.define) {
+    var defs = parsedArgs.define;
+    defs.forEach(function (def) {
+        try {
+            var match = /^(([^=]+)=(.+)|@(.*))$/.exec(def);
+            if (match) {
+                if (!match[4]) {
+                    var val = JSON.parse(match[3]);
+                    setProperty(settings, match[2], val);
+                }
+                else {
+                    var obj = fs.readJsonSync(match[4]);
+                    Object.entries(obj).forEach(([key, val]) => {
+                        settings[key] = val;
+                    });
+                }
+            }
+            else {
+                throw new Error("Unexpected option: "+def);
+            }
+        }
+        catch (e) {
+            console.log(e);
```

Comment:
I think we need to better define the behaviour if any of the properties are invalid.

Currently, if one of the properties specifies a file to load, if the file is invalid, it will console.log an error, stop parsing any of the remaining defines, but it then continues to start Node-RED.

I think if any of the defines are invalid, it should log an appropriate error and *not* continue starting Node-RED.

Reply:
Fixed Node-RED stop processing when error occurs.